src/learner.cc

```diff
@@ -489,13 +489,13 @@ class LearnerConfiguration : public Learner {
       CHECK(matrix.first);
       CHECK(!matrix.second.ref.expired());
       const uint64_t num_col = matrix.first->Info().num_col_;
+      LOG(WARNING) << "Configuring features for rank " << rabit::GetRank()
+                   << ", got " << num_col << " features.";
       CHECK_LE(num_col, static_cast<uint64_t>(std::numeric_limits<unsigned>::max()))
           << "Unfortunately, XGBoost does not support data matrices with "
           << std::numeric_limits<unsigned>::max() << " features or greater";
       num_feature = std::max(num_feature, static_cast<uint32_t>(num_col));
     }
-    // run allreduce on num_feature to find the maximum value
-    rabit::Allreduce<rabit::op::Max>(&num_feature, 1, nullptr, nullptr, "num_feature");
```

Comment:
I thought this was needed, because when workers read from a LIBSVM file, they may have different `num_feature`.

Reply:
It's actually being done in all DMatrix constructors. The only reason it hasn't worked on Spark is because this construction does not run inside of a rabit context.